### PR TITLE
chore(security): Add slither static analysis

### PR DESCRIPTION
**Pull request by dgusakov**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+eth-abi>=2.1.1
```
Comment (TheDZhon) [inline requirements.txt:1]: why do we need `requirements.txt` if `pyproject.toml` is already here?

File: .github/workflows/slither.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
+name: Slither Analysis
+
+on:
+  push:
+    branches: [develop, master]
+  pull_request:
+    branches: [develop]
```
Comment (TheDZhon) [inline .github/workflows/slither.yml:7]: why `master` is omitted?

File: slither.config.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+    "exclude_informational": true,
+    "exclude_low": true,
+    "exclude_medium": false,
+    "exclude_high": false,
+    "filter_paths": "(.*test.*/|.*template/|.*mocks/|node_modules/|.*brownie/)"
```
Comment (TheDZhon) [inline slither.config.json:6]: please, add also './dependencies' subfolder with all possible contents inside

**Review by TheDZhon — CHANGES_REQUESTED** [review]
LGTM :+1: , only some minors to discuss

File: .github/workflows/slither.yml (hunk 2 above)
Comment (dgusakov) [reply .github/workflows/slither.yml:7]: Because the report will be generated on push to develop

File: requirements.txt (hunk 1 above)
Comment (dgusakov) [reply requirements.txt]: Removed

File: slither.config.json (hunk 3 above)
Comment (dgusakov) [reply slither.config.json]: Added

**Comment by TheDZhon** [comment]
Merged into `develop` via CLI interaction to address merge conflicts.

**Pull request merged**